### PR TITLE
#7 - [FA] Sign Up - Petar

### PR DESCRIPTION
**Pull request by Nikolov87**
## Naming Rules
Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description
Write a summary of your change. Explain why you have made it.
Creating and sailing Sign Up form

## Related to
#7

## Checklist:
- [ x ] My code follows the style guidelines of this project
- [ x ] I have carefully reviewed my own code
- [ x ] I have commented my code
- [ x ] I have updated any documentation

### CODE REVIEW

**Review by Johng117 — APPROVED** [review]
Good Job Petar.

File: client/src/Pages/Signup/SignupForm.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [SelectDepartment, setRole] = useState("SelectDepartment"); // Default role is "admin"
+	const navigate = useNavigate();
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		navigate("/");
+
+		// Perform your signup logic here using the state variables
```
Comment (ChidimmaOfodum) [inline client/src/Pages/Signup/SignupForm.js:18]: The signup data should be sent to the backend with a post request. The logic is missing.

File: client/src/Pages/Signup/SignupForm.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		console.log("Form submitted with role:", SelectDepartment);
+	};
+
+	const redirectPath =
+		SelectDepartment === "admin" ? "/admin-dashboard" : "/staff-dashboard";
```
Comment (ChidimmaOfodum) [inline client/src/Pages/Signup/SignupForm.js:23]: Staff in HR department are admins

File: client/src/Pages/Signup/SignupForm.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				<div className="mb-3 row">
+					<div className="col-sm-10">
+						<select
+							className="form-control"
+							value={SelectDepartment}
+							onChange={(e) => setRole(e.target.value)}
+						>
+							<option value="department">Department </option>
+
+							<option value="staff" href="#">
```
Comment (ChidimmaOfodum) [inline client/src/Pages/Signup/SignupForm.js:69]: The department data will be fetched from the api and displayed here

**Review by ChidimmaOfodum — CHANGES_REQUESTED** [review]

**Comment by ChidimmaOfodum** [comment]
Thank you Petar. Also remember to pull the recent changes from the dev branch before making a pull request

File: client/src/Pages/Signup/SignupForm.js (hunk 3 above)
Comment (Nikolov87) [reply client/src/Pages/Signup/SignupForm.js]: Thank you Chidimma, this is fixed :)

**Pull request merged**